bookkeeper-server/src/main/java/org/apache/bookkeeper/http/GetLastLogMarkService.java

```diff
@@ -0,0 +1,104 @@
+import com.google.common.collect.Maps;
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import org.apache.bookkeeper.bookie.Journal;
+import org.apache.bookkeeper.bookie.LedgerDirsManager;
+import org.apache.bookkeeper.bookie.LogMark;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.http.service.HttpService;
+import org.apache.bookkeeper.http.service.HttpServiceRequest;
+import org.apache.bookkeeper.http.service.HttpServiceResponse;
+import org.apache.bookkeeper.util.DiskChecker;
+import org.apache.bookkeeper.util.JsonUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpService that handle Bookkeeper Configuration related http request.
+ */
+public class GetLastLogMarkService implements HttpService {
+
+    static final Logger LOG = LoggerFactory.getLogger(GetLastLogMarkService.class);
+
+    protected ServerConfiguration conf;
+
+    public GetLastLogMarkService(ServerConfiguration conf) {
+        Preconditions.checkNotNull(conf);
+        this.conf = conf;
+    }
+
+    @Override
+    public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
+        HttpServiceResponse response = new HttpServiceResponse();
+
+        if (HttpServer.Method.GET == request.getMethod()) {
+            try {
+                /**
+                 * output:
+                 *  {
+                 *    "<Journal_id>" : "<Pos>",
+                 *    ...
+                 *  }
+                 */
+                Map<String, String> output = Maps.newHashMap();
+
+                List<Journal> journals = Lists.newArrayListWithCapacity(conf.getJournalDirs().length);
+                for (File journalDir : conf.getJournalDirs()) {
+                    journals.add(new Journal(journalDir, conf, new LedgerDirsManager(conf, conf.getLedgerDirs(),
+                      new DiskChecker(conf.getDiskUsageThreshold(), conf.getDiskUsageWarnThreshold()))));
+                }
+                for (Journal journal : journals) {
+                    LogMark lastLogMark = journal.getLastLogMark().getCurMark();
+                    LOG.debug("LastLogMark: Journal Id - " + lastLogMark.getLogFileId() + "("
+                      + Long.toHexString(lastLogMark.getLogFileId()) + ".txn), Pos - "
+                      + lastLogMark.getLogFileOffset());
+                    output.put("LastLogMark: Journal Id - " + lastLogMark.getLogFileId()
+                        + "(" + Long.toHexString(lastLogMark.getLogFileId()) + ".txn)",
+                        "Pos - " + lastLogMark.getLogFileOffset());
+                }
+
+                String jsonResponse = JsonUtil.toJson(output);
+                LOG.info("output body:" + jsonResponse);
+                response.setBody(jsonResponse);
+                response.setCode(HttpServer.StatusCode.OK);
+                return response;
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
remove

Reply:
Thanks. will change to LOG.error as Enrico suggested.